src/wazuh_modules/wm_sca.c

```diff
@@ -685,13 +688,30 @@ static int wm_sca_check_policy(cJSON *policy, cJSON *profiles) {
                 mwarn("Check ID not found.");
                 free(read_id);
                 return retval;
-            } else if (check_id->valueint <= 0) {
+            }
+            if (check_id->valueint <= 0) {
                 // Invalid ID
                 mwarn("Invalid check ID: %d", check_id->valueint);
                 free(read_id);
                 return retval;
             }
 
+            char *coincident_policy;
+            char *key_id;
+            size_t key_length = snprintf(NULL, 0, "%d", check_id->valueint);
+            os_malloc(key_length + 1, key_id);
+            snprintf(key_id, key_length + 1, "%d", check_id->valueint);
+
+            if((coincident_policy = (char *)OSHash_Get(global_check_list, key_id)), coincident_policy){
+                // Invalid ID
+                mwarn("Invalid check ID: %d. It was found in policy %s", check_id->valueint, coincident_policy);
```

Comment:
Confusing warning message, "invalid" can mean a lot of things... Let's change it to `"Duplicated check ID: %d. First appearance at policy '%s'`